packages/financial-templates-lib/src/price-feed/utils.ts

```diff
@@ -199,41 +199,70 @@ export const PriceHistory = <T>(
   };
 };
 
-/**
- * @notice Constructs the BlockFinder, which is a simple class-like object to find blocks by timestamp, but keeps a
- *      cache to optimize the search.
- * @param {Function} requestBlock async function, like web3.eth.getBlock that returns a block for a particular block
- *      number or returns the latest block if no argument is provided. Blocks returned must have a `number` and
- */
-export const BlockFinder = <T extends { number: number; timestamp: number | string }>(
-  requestBlock: (requestedBlock: string | number) => Promise<T>,
-  blocks: T[] = []
-): {
-  getBlockForTimestamp: (timestamp: number) => Promise<T>;
-} => {
-  assert(requestBlock, "requestBlock function must be provided");
+export class BlockFinder<T extends { number: number; timestamp: number | string }> {
+  constructor(
+    private readonly requestBlock: (requestedBlock: string | number) => Promise<T>,
+    private readonly blocks: T[] = []
+  ) {
+    assert(requestBlock, "requestBlock function must be provided");
+  }
+
+  /**
+   * @notice Gets the latest block whose timestamp is <= the provided timestamp.
+   * @param {number} timestamp timestamp to search.
+   */
+  public async getBlockForTimestamp(timestamp: number | string): Promise<T> {
```

Comment:
this is good and will work well for blocks close to the current timestamp, which is the situation we are in for the relayer.